### PR TITLE
Added Delaunay related functions to processing module

### PR DESCRIPTION
**Pull request by alex-graciano**
Implemented the Delaunay related functions in `processing` module.

They have been implemented using the Voronoi diagram dual, so the turf.js functions are used at low level. Turf.js returns a triangle soup for its voronoi so a previous step to build topology is required. It has been done by means a ugly triple self joint

```sql
        SELECT ARRAY( SELECT ST_MAKELINE([p1.dual.point, p2.dual.point, p3.dual.point, p1.dual.point]) AS triangle
                    FROM triplets AS p1
                    JOIN triplets AS p2
                    ON ST_TOUCHES(p1.dual.poly, p2.dual.poly)
                    AND ST_EQUALS(p1.dual.point, p2.dual.point) = FALSE
                    JOIN triplets AS p3
                    ON ST_TOUCHES(p1.dual.poly, p3.dual.poly) AND ST_TOUCHES(p2.dual.poly, p3.dual.poly)
                    AND ST_EQUALS(p2.dual.point, p3.dual.point) = FALSE
                    AND p1.dual.geoid <= p2.dual.geoid AND p2.dual.geoid <= p3.dual.geoid )
```
If anyone comes up with a better idea it will be greatly received.

In contrast with the original PostGIS function which uses a flag to choose the flavour (lines/polygons), I am using two different functions in order to follow the same structure we use for the Voronois'.

Also, some code file restructuring in the module have been made to avoid dependency issues among functions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,15 @@ This document describes changes between tagged releases of the CARTO Spatial Ext
 
 Not all changes in the project will be included here, but it should mention any new features and user-oriented bug fixes. You can see the full set of change between releases by browsing the git commits.
 
+## 2021-05-04
+
+* Add the processing module (1.0.0)
+* Add functions to processing module
+
+## 2021-05-03
```
Comment (vdelacruzb) [inline CHANGELOG.md:11]: Can you remove 2021-05-03.
And put `* Add the accessors module (1.0.0)` together with processing?

File: processing/CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,11 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [1.0.0] - 2021-04-30
+## [1.0.0] - 2021-05-04
 
 ### Added
 * Initial implementation of the module.
 * Add ST_VORONOIPOLYGONS function based on buffer (turfjs v6.3.0).
 * Add ST_VORONOILINES function based on buffer (turfjs v6.3.0).
+* Add ST_DELAUNAYPOLYGONS function.
```
Comment (vdelacruzb) [inline processing/CHANGELOG.md:13]: Here, maybe @Jesus89  can give us advice on what to do. Do we go for 1.1.0 2021-05-04 or put everything together?

Comment (Jesus89) [reply processing/CHANGELOG.md:13]: Yes. I have changed the base branch to merge this one in Voronoi's PR. But the rule is: releases on the same day will be included in the same version number (in this case 1.0.0).

File: CHANGELOG.md (hunk 1 above)
Comment (alex-graciano) [reply CHANGELOG.md]: Sure!

**Comment by Jesus89** [comment]
Once this is OK, we can merge it in Voronoi's PR and make a full review there.

File: processing/CHANGELOG.md (hunk 2 above)
Comment (alex-graciano) [reply processing/CHANGELOG.md:13]: Ok, let's wait

**Review by Jesus89 — APPROVED** [review]
LGTM

**Review by vdelacruzb — APPROVED** [review]
LGTM!

**Pull request merged**